src/StackExchange.Redis/ConnectionMultiplexer.cs

```diff
@@ -2016,6 +2016,21 @@ public void Dispose()
             oldTimer?.Dispose();
         }
 
+        /// <summary>
+        /// Release all resources associated with this object.
+        /// </summary>
+        public async ValueTask DisposeAsync()
+        {
+            GC.SuppressFinalize(this);
+            await CloseAsync(!_isDisposed);
+            if (sentinelConnection is not null)
+            {
+                await sentinelConnection.DisposeAsync();
+            }
```

Comment:
This can race, where `sentinelConnection` is nulled out between the `if` and the `await`. Instead, what you want to do is use the `if (sentinelConnection is ConnectionMultiplexer sentinel) { await sentinel.DisposeAsync(); }` pattern so that you're eliminating the potential race and null ref (more common than you'd think in disposing!)

Reply:
Well spotted. I guess it could also be `await (sentinel?.DisposeAsync() ?? default);` to match the synchronous implementation a little closer, if you prefer? I just don't find it quite as readable.

Reply:
Would stick with the `if`, you don't want to pay state machine overhead awaiting if we know there's nothing to await :)

Reply:
I don't think it makes any difference. To the best of my knowledge, the state machine isn't initialized if the (Value)Task is completed synchronously (the `if (!awaiter.IsCompleted)` [here](https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQAwAIpwKwG5lQDMmM6AwugN7Lq2bFQAcmAbOgJIBmAFAJRU06Q9gEEAzgE8AdgGMAIgEsxABwD2YgIYAjADYBTTI0Ur12/egC86KQFcdO/EiFCFndNybG1m3QaUdxaXklbzMDTxDTXzh+aidnBKgATkMvKP04ADo0sT1A2T5HBNoAX0FS8vpDVnQ5PU4NOwAXPgF4hNFJWRyw1MifcytbeyLi5PcIkwG9AH5s/ryumVaZmfRgesadJt5RiqQSoA===)), but I'm not too knowledgeable on the async/await internals.

In any case, I've pushed the version you suggested. 😄